src/L2GovernorMetadata.sol

```diff
@@ -28,7 +38,21 @@ abstract contract L2GovernorMetadata {
     virtual
   {
     _proposals[proposalId] = Proposal(voteStart, voteEnd, isCanceled);
-    emit ProposalAdded(proposalId, voteStart, voteEnd, isCanceled);
+    if (isCanceled) {
+      emit ProposalCanceled(proposalId);
+    } else {
+      emit ProposalCreated(
+        proposalId,
+        address(0),
+        new address[](0),
+        new uint256[](0),
+        new string[](0),
+        new bytes[](0),
+        voteStart,
+        voteEnd,
+        ""
```

Comment:
Instead of an empty string here, perhaps we can emit something helpful, like `"Mainnet proposal ${proposalId}"`?

Reply:
done!